draft-ietf-tls-esni.md

```diff
@@ -662,15 +673,24 @@ extension across all application profiles. The target padding length of most
 ClientHello extensions can be computed in this way.
 
 In contrast, clients do not know the longest SNI value in the client-facing
-server's anonymity set without server input. For the "server_name" extension
-with length D, clients SHOULD use the server's length hint L
-(ECHConfig.contents.maximum_name_length) when computing the padding as follows:
-
-1. If L >= D, add L - D bytes of padding. This rounds to the server's
-   advertised hint, i.e., ECHConfig.contents.maximum_name_length.
-2. Otherwise, let P = 31 - ((D - 1) % 32), and add P bytes of padding, plus an
-   additional 32 bytes if D + P < L + 32. This rounds D up to the nearest
-   multiple of 32 bytes that permits at least 32 bytes of length ambiguity.
+server's anonymity set without server input. Clients SHOULD use the ECHConfig's
+`maximum_name_length` field as follows, where L is the `maximum_name_length`
+value.
+
+1. If the ClientHelloInner contained a "server_name" extension with a name of
+   length D, add max(0, L - D) bytes of padding.
+2. If the ClientHelloInner did not contain a "server_name" extension (e.g., if
+   the client is connecting to an IP address), add L + 7 bytes of padding. This
+   is the length of a "server_name" extension with an L-byte name.
+
+Finally, the client SHOULD pad the entire message as follows:
+
+1. Let L be the length of the EncodedClientHelloInner with all the padding
+   computed so far.
+2. Let P = 31 - ((L - 1) % 32) and add P bytes of padding.
+
+This rounds the overall structure up to a multiple of 32 bytes. This reduces
+the range of lengths of the EncodedClientHelloInner across all clients.
```

Comment:
```suggestion
This rounds the length of EncodedClientHelloInner up to a multiple of 32 bytes,
reducing lengths of this structure across all clients.
```

Reply:
Agreed on the first half. I worry "reducing lengths" reads like making the lengths smaller. I guess "range of lengths" also has a different meaning... I was thinking the range of a function. :-)

How about "reducing the set of possible lengths across all clients".

Reply:
That works for me too 👍

Reply:
Done.